### PR TITLE
feat: Leaderboard

### PR DESCRIPTION
**Pull request by admsk2**
https://lifi.atlassian.net/browse/LF-8129

### CODE REVIEW

File: src/components/ProfilePage/Leaderboard/Leaderboard.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    <LeaderboardContainer>
+      <NoSelectTypography fontSize="14px" lineHeight="18px" fontWeight={700}>
+        RANK
+      </NoSelectTypography>
+      <NoSelectTypography
+        color={theme.palette.mode === 'light' ? theme.palette.accent1.main : theme.palette.white.main}
+        fontWeight={700}
+        sx={{ fontSize: { xs: 28, sm: 48 } }}
+      >
+        {leaderboardUserData ? leaderboardUserData.position : '-'}
```
Comment (oktapodia) [inline src/components/ProfilePage/Leaderboard/Leaderboard.tsx:45]: Can be simplify with 
```suggestion
        {leaderboardUserData?.position ?? '-'}
```

File: src/components/ProfilePage/Leaderboard/Leaderboard.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      </NoSelectTypography>
+      <Stack direction={'column'} sx={{ margin: '20px 0' }}>
+        {!leaderboardData?.length ? <LeaderboardSkeleton />
+          : leaderboardData?.map((entry: any, index: number) => (
```
Comment (oktapodia) [inline src/components/ProfilePage/Leaderboard/Leaderboard.tsx:49]: use type instead please

**Review by oktapodia — APPROVED** [review]
Few comments but not blocking

**Review by oktapodia — APPROVED** [review]
LGTM

**Review by dennyscode — APPROVED** [review]

**Pull request merged**